docs/troubleshooting.md

```diff
@@ -99,6 +99,108 @@ for p in $(oc get pods -l component=es -o jsonpath={.items[*].metadata.name}); d
 done
 ```
 
+## Fluentd
+### Fluentd is holding onto deleted journald files that have been rotated
+This [issue](https://bugzilla.redhat.com/show_bug.cgi?id=1664744) causes `/var/log` to fill up with file handles of 
+deleted journald files.  The only known solution at this time is to periodically cycle Fluentd.  An OKD `CronJob` can be added to perform this function, for example, by checking every 10 minutes the amount of available space to fluentd:
+```
+apiVersion: v1
+kind: Template
+metadata:
+  name: fluentd-reaper
+objects:
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: Role
+  metadata:
+    name: fluentd-reaper
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - pods
+    verbs:
+    - delete
+  - apiGroups:
+    - ""
+    resources:
+    - pods/exec
+    verbs:
+    - create
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: fluentd-reaper
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: Role
+    name: fluentd-reaper
+  subjects:
+  - kind: ServiceAccount
+    name: aggregated-logging-fluentd
+    namespace: openshift-logging
+- apiVersion: batch/v1beta1
+  kind: CronJob
+  metadata:
+    name: fluentd-reaper
+    labels:
+       provider: openshift
+       logging-infra: fluentd-reaper
+  spec:
+    schedule: "${REAP_SCHEDULE}"
+    jobTemplate:
+      spec:
+        template:
+          metadata:
+            labels:
+              provider: openshift
+              logging-infra: fluentd-reaper
+          spec:
+            serviceAccount: aggregated-logging-fluentd
+            serviceAccountName: aggregated-logging-fluentd
+            containers:
+            - env:
+              - name: ENABLED
+                value: "true"
+              - name: REAP_THRESHOLD
+                value: "${REAP_THRESHOLD}"
+              name: cli
+              image: ${CLI_IMAGE}
+              command: ["/bin/bash", "-c"]
+              args:
+                - if [ "$ENABLED" != "true" ]; then
+                    echo "Job is disabled; exiting early";
+                    exit 0;
+                  fi;
+                  echo "Checking fluentd pods for space issues on /var/log...";
+                  pods=$(oc get pods -l component=fluentd -o jsonpath={.items[*].metadata.name});
+                  for p in $pods; do
+                    echo "Checking $p...";
+                    space=$(oc exec -c fluentd-elasticsearch $p -- bash -c 'df -Ph /var/log | tail -1 | cut -s -d " " -f 12 | rev | cut -b 2- | rev');
```

Comment:
Any way to make this more specific to our use case?  If /var/log is filling up in general it seems like we'll take down all pods even though we may not be the cause of it filling.  Maybe that doesn't matter since we'd run out of space anyway but may be a red herring for debugging.

Reply:
It would be nice if we could check the file descriptors but then we need to build `lsof` (?) into the image and then we could check for the deleted file handles (unless there is some other way).   I have general concerns with using disk space as the threshold; I can imagine a case where we have high log volume that is not being ingested fast enough into Elasticsearch.  It could be that we require a lot of disk to buffer (maybe exceeding the threshold) but that we gain nothing by cycling fluent because its not going to release any file handles that point to files of significant size.  This would put us into an undesired loop.

Reply:
If there are no adverse affects why not just cycle the pods every X minutes of uptime to clear open handles? Does that eliminate the hot loop?